spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreOperations.java

```diff
@@ -131,7 +132,14 @@
 	 * @return a list of the objects found. If no keys could be found the list will be
 	 * empty.
 	 */
-	<T> Iterable<T> query(Query<Entity> query, Class<T> entityClass);
+	<T> Iterable<T> query(Query query, Class<T> entityClass);
```

Comment:
We cannot be sure if the `query` is a key-query or that it returns entities. Technically it would need to be a `StructredQuery<Key>` , but we cannot check the inner type of the StructuredQuery to make sure. (note that `KeyQuery` mentioned in the next method `queryKeys` is a child type of `StructuredQuery<Key>`, but it's not necessarily the only child). 

Even if we could check that inner type, we would still be requiring the user to make sure he has the right type of query AND the right type of entityClass. I think in that case he might as well just use `queryKeys` where he doesn't need to provide the entityclass at all.